### ISSUE TITLE
allow import of ed25519 seed

### ISSUE THREAD

**Issue by RoanBrand** [polkadot-js/extension#888]
I participated in Polkadot crowdloans via https://parallel.fi/ using the extension tied to a Ledger x.

Now the Moonbeam projects wants us to sign an attestation on their site (https://parallelfinance.medium.com/attention-51de62ae3ba7)
as well.

But I cannot seem to sign a message:
![image](https://user-images.githubusercontent.com/14920090/141308044-968064cb-ca11-4dcd-9045-ce796d2cd68d.png)

I wanted to find out how I can temporarily export a private key of some sorts from my ledger's mnemonic, and import that in the extension, but the best I seem to be able to get out with the subkey tools is a hex ed25519 seed.

I cannot seem to get 12 or 24 words from this that I can import to the extension to get the same address. Any ideas?

Thanks!

**Comment by jacogr** [comment]
The Ledger doesn't have a bytes-signing interface. Additionally, not sure that the Acala team (or Parallel) has a Ledger app available, so that may create issues.

Generally via the apps UI you can import the seed in any crypto and export to JSON - the JSON imports (any crypto, e.g. ed25519 allowed) will work in the extension. Obviously you need to ensure the addresses do match, Ledger will generate a different address for each network from the same seed. 

Apps UI does have the capability to import Ledger mnemonics, when connected to a Ledger-capabale envisionment, i.e. on e.g. Kusama/Polkadot which does have Ledger apps, you can generate a standard JSON pair from the mnemonic, there is an option when connected to one of these chains. (There is also a cli utility so it doesn't need to be imported anywhere as mnemonic, e.g. https://github.com/jacogr/sample-ledger-ed25519)

TL:DR To keep the extension simple, only sr25519 is available via mnemonics. However with JSON imports any crypto is supported.

**Comment by RoanBrand** [comment]
I need to sign a Moonbeam attestation message using the polkadot address and extension, with ledger. Doesn't seem to be able.

I have a ed25519 that I got from using subkey that seems to generate the same Polkadot address I see in the extension when connecting my ledger. The problem is I need a reverse conversion now from that seed to a mnemonic I can import in the extension to effectively create a software account with the same address as my ledger one to get around this problem. Seems impossible.

Now I am screwed because I participated in the crowdloans via parallel.fi using ledger, which worked, but now I cannot seem to sign a simple message for Moonbeam, which requires it.

Any help would be appreciated thanks.

**Comment by RoanBrand** [comment]
![image](https://user-images.githubusercontent.com/14920090/141313264-002bb0c2-c1bb-448a-8f85-71989677ce61.png)

Firefox doesn't seem to be supported either, although it claims to be.

**Comment by jacogr** [comment]
FF doesn't do Ledger, it doesn't have WebUsb or WedHid.

**Comment by RoanBrand** [comment]
I got around this by using the polkadot desktop wallet https://github.com/polkadot-js/apps/releases
as it supports import of ed25519 seed. Then I exported the json file from there, import it into the polkadot extension. Then now if sign message:
![image](https://user-images.githubusercontent.com/14920090/141319620-b2697e92-3ec3-4430-a825-1bfc5d1a9820.png)

Guess you guys need to work on Ledger support.

**Comment by jacogr** [comment]
There is nothing I can do with "Ledger support" - it only supports signing transactions, not raw messages. If you have the JSON and the Ledger added with the same address, only the last one will be used. Hence you needing a password now.
